core/controllers/contributor_dashboard.py

```diff
@@ -260,6 +263,45 @@ def _content_in_review(self, state_name, content_id, suggestions):
             s.change.content_id == content_id for s in suggestions)
 
 
+class MachineTranslatedTextHandler(base.BaseHandler):
+    """Provide a machine translation of exploration content."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+
+    @acl_decorators.open_access
+    def get(self):
+        """Handles GET requests."""
+        exp_id = self.request.get('exp_id')
+        state_name = self.request.get('state_name')
+        content_ids = json.loads(self.request.get('content_ids'))
+        target_language_code = self.request.get('target_language_code')
+
+        # TODO(#12341): Tidy up this logic once we have a canonical list of
+        # language codes.
+        if not utils.is_supported_audio_language_code(
```

Comment:
Is there a need to check first that target_language_code (and exp_id, state_name for that matter) are not None?

Reply:
I believe not, `NoneType` should fail the type validations just like any other type. Is this worth adding tests for?

Reply:
Yup, worth adding tests. Thanks!

Reply:
I added a variety of tests. PTAL.